app/controllers/api/v1/rides_controller.rb

```diff
@@ -1,15 +1,39 @@
 class Api::V1::RidesController < ApiController
-
+  before_action :authorize_admin, only: [:create]
+  
   def index
     render json: Ride.all
   end
 
+  def create
+    ride = Ride.new(ride_params)
+    if ride.save
+      render json: ride 
+    else
+      render json: {errors: ride.errors.full_messages.to_sentence}, status: 401
+    end
+  end
+
   def show   
     render json: Ride.find(params[:id]), serializer: RideShowSerializer, include: ['reviews.user']
   end
 
   def search
     render json: Ride.where("name ILIKE ?", "%#{params['search_string']}%")
   end
-  
+
+  def authorized  
+  end
+
+  protected
+
+  def authorize_admin
+    if !user_signed_in? || !(current_user.role == "admin")
+      render json: {error: ["Only admins have access to this feature"]}
+    end
+  end
```

Comment:
rather than having to define this `authorize_admin` method for every controller, we should add this to the ApiController under a protected method so that it can be inherited and used by any controller that inherits from the ApiController

then, our strong params should be defined under `private` so that they are only accessible from this controller